sphinx/api/configuration/file-creation.rst

```diff
@@ -25,6 +25,42 @@ root directory to install file targets with a relative path:
       :opt svn: The Subversion scheme. The location is a Subversion URL or
          an FCM location keyword. A URI with these schemes ``svn``,
          ``svn+ssh`` and ``fcm`` are automatically recognised.
+      :opt git: The Git scheme. The location is complex due to Git semantics.
+         It must have the scheme ``git`` and be of the form
+         ``git:REPOSITORY_URL::PATHSPEC::TREEISH``. ``REPOSITORY_URL`` should
+         be a Git repository URI which may itself have a scheme ``ssh``,
+         ``git``, ``https``, or be of the form ``HOST:PATH``, or ``PATH`` for
+         local repositories. ``PATHSPEC`` should be a path to a file or
+         directory that you want to extract. ``TREEISH`` should be a tag,
+         branch, or long commit hash to specify the commit at which you want
+         to extract. These should follow the same semantics as if you git
+         cloned ``REPOSITORY_URL``, git checkout'ed ``TREEISH``, and extracted
+         the path ``PATHSPEC`` within the clone. ``PATHSPEC`` must end with a
+         trailing slash (``/``) if it is a directory. To extract from the root
+         of the repository use a ``PATHSPEC`` of ``./`` e.g.
+         ``git:git@github.com:metomi/rose::./::2.2.0``. It may help to think
+         of the parts of the location as git:Where::What::When. Examples:
```

Comment:
[minor] Broke the paragraph up with bullet points and co-located the `PATHSPAC` stuff (no text alterations):

```suggestion
      :opt git: The Git scheme. The location is complex due to Git semantics.
         It must have the scheme ``git`` and be of the form
         ``git:REPOSITORY_URL::PATHSPEC::TREEISH``:
         
         * ``REPOSITORY_URL`` should
           be a Git repository URI which may itself have a scheme ``ssh``,
           ``git``, ``https``, or be of the form ``HOST:PATH``, or ``PATH`` for
           local repositories.
         * ``PATHSPEC`` should be a path to a file or
           directory that you want to extract.
           The ``PATHSPEC`` must end with a
           trailing slash (``/``) if it is a directory. To extract from the root
           of the repository use a ``PATHSPEC`` of ``./`` e.g.
           ``git:git@github.com:metomi/rose::./::2.2.0``.
         * ``TREEISH`` should be a tag,
           branch, or long commit hash to specify the commit at which you want
           to extract.

         These should follow the same semantics as if you git
         cloned ``REPOSITORY_URL``, git checkout'ed ``TREEISH``, and extracted
         the path ``PATHSPEC`` within the clone. It may help to think
         of the parts of the location as ``git:Where::What::When``.
         
         ..rubric:: Examples:
```